Scripts/Runtime/Entities/TaskSystem/TaskData/DataStream/Abstract/AbstractDataStream.cs

```diff
@@ -0,0 +1,82 @@
+using Anvil.CSharp.Core;
+using Anvil.CSharp.Logging;
+using Anvil.Unity.DOTS.Jobs;
+using System;
+using System.Reflection;
+using Unity.Jobs;
+#if DEBUG
+using Unity.Profiling;
+using Unity.Profiling.LowLevel;
+#endif
+
+#if ANVIL_DEBUG_LOGGING_EXPENSIVE
+using Unity.Collections;
+#endif
+
+namespace Anvil.Unity.DOTS.Entities.Tasks
+{
+    internal abstract class AbstractDataStream : AbstractAnvilBase
+    {
+        public static readonly BulkScheduleDelegate<AbstractDataStream> CONSOLIDATE_FOR_FRAME_SCHEDULE_FUNCTION = BulkSchedulingUtil.CreateSchedulingDelegate<AbstractDataStream>(nameof(ConsolidateForFrame), BindingFlags.Instance | BindingFlags.NonPublic);
+
+        private readonly AbstractTaskDriver m_OwningTaskDriver;
+        private readonly AbstractTaskSystem m_OwningTaskSystem;
+
+        public Type Type { get; }
+
+        public AccessController AccessController { get; }
+
+#if DEBUG
+        protected ProfilerMarker Debug_ProfilerMarker { get; }
+#endif
+#if ANVIL_DEBUG_LOGGING_EXPENSIVE
+        protected FixedString128Bytes Debug_DebugString { get; }
+#endif
+
+        protected AbstractDataStream(AbstractTaskDriver taskDriver, AbstractTaskSystem taskSystem)
+        {
+            Type = GetType();
+            AccessController = new AccessController();
+            m_OwningTaskDriver = taskDriver;
+            m_OwningTaskSystem = taskSystem;
+
+
+#if DEBUG
+            Debug_ProfilerMarker = new ProfilerMarker(ProfilerCategory.Scripts, ToString(), MarkerFlags.Script);
+#endif
+#if ANVIL_DEBUG_LOGGING_EXPENSIVE
+            Debug_DebugString = new FixedString128Bytes(ToString());
+#endif
+        }
+
+        protected sealed override void DisposeSelf()
```

Comment:
I'm not sure about sealing `DisposeSelf`

Why not require each subclass to acquire before they're going to call dispose on their data type(s)? This allows others to implement dispose logic beyond just data streams.

Reply:
In my mind it was so that anyone making derived DataStreams didn't have to remember to do that acquisition. It felt much safer to acquire for disposal, allow the derived classes to clean up and then dispose the access controller.

Reply:
hmmm, yea I'm just worried about constraining other disposal logic. Maybe we just override `Dispose`?

I don't think it's a big deal to ask subclasses to acquire before they perform a write operation but see where you're coming from.